include/rfr/forests/regression_forest.hpp

```diff
@@ -351,6 +356,82 @@ class regression_forest{
 		return(rv);
 	}
 
+    /* \brief Collects the response values of the trees for a set of instance feature vectors and then
+     *        for each tree takes the average over all the collected response values.
+     *
+     *        In the case of log transformation the response values are decompressed before averaging.
+	 *
+	 * \param feature_matrix: nested vector where each internal vector is a feature vector
+	 * \param log_y: boolean which determines if there should be corrected for log transforms
+	 *
+	 * \return The predicted cost marginalized over all the input vectors for each tree in the forest.
+	 */
+	std::vector<num_t> predict_marginalized_over_instances(const std::vector<std::vector<num_t>> &feature_matrix, const bool log_y = false) const{
+	    std::vector<num_t> tree_values(the_trees.size(), 0.0);
+	    int tree_id, entry_counter;
+		for (tree_id = the_trees.size()-1; tree_id >= 0; tree_id--){
+		    entry_counter = 0;
+	        for (auto &feature_vector: feature_matrix){
+	            for (auto val: the_trees[tree_id].leaf_entries(feature_vector)){
+	                tree_values[tree_id] += log_y ? std::exp(val) : val;
```

Comment:
There is a risk of overflowing. Maybe you could compute the mean like this: https://stackoverflow.com/a/1934266